### ISSUE TITLE
kustomize build does not preserve filename lexical order

### ISSUE THREAD

**Issue by lionelvillard** [kubernetes-sigs/kustomize#491]
one of the `kubectl` features is to apply configurations sequentially following filename lexical order.  This is useful for instance to create a namespace before applying the rest of the configuration. `kustomize build` breaks this feature as it generates a single specification. 

One possible solution is to allow `kustomize build -o directory/ && kubectl apply -f directory/`. It's not as nice as piping the configuration but at least that would solve the problem stated above.

**Comment by Liujingfang1** [comment]
kustomize build output has a predefined order to handle this concern. The output will follow this order
```
	"Namespace",
	"CustomResourceDefinition",
	"ServiceAccount",
	"Role",
	"ClusterRole",
	"RoleBinding",
	"ClusterRoleBinding",
	"ConfigMap",
	"Secret",
	"Service",
	"Deployment",
	"StatefulSet",
	"CronJob",
	"PodDisruptionBudget",
```
If it doesn't satisfy your usage, let us know which order is missing.

**Comment by lionelvillard** [comment]
is this implemented in v1.0.8? That should work except for CRD instances (not mentioned in your list), as it takes some time for the CRDs to be online.

**Comment by lionelvillard** [comment]
thanks for the explanation. It's working as expected after fixing a typo.
